components/data_prefetch/index.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {ActionCreatorsMapObject, bindActionCreators, Dispatch} from 'redux';
+
+import {getCurrentChannelId, getUnreadChannels} from 'mattermost-redux/selectors/entities/channels';
+import {memoizeResult} from 'mattermost-redux/utils/helpers';
+import {isChannelMuted} from 'mattermost-redux/utils/channel_utils';
+import {getMyChannelMemberships} from 'mattermost-redux/selectors/entities/common';
+
+import {Channel, ChannelMembership} from 'mattermost-redux/types/channels';
+
+import {ActionFunc, GenericAction} from 'mattermost-redux/types/actions';
+import {RelationOneToOne} from 'mattermost-redux/types/utilities';
+
+import {GlobalState} from 'types/store';
+
+import {prefetchChannelPosts} from 'actions/views/channel';
+
+import DataPrefetch from './data_prefetch';
+
+type Actions = {
+    prefetchChannelPosts: (channelId: string, delay: number | undefined) => Promise<{data: {}}>;
+};
+
+const prefetchQueue = memoizeResult((channels: Channel[], memberships: RelationOneToOne<Channel, ChannelMembership>) => {
+    return channels.reduce((acc: Record<string, string[]>, channel: Channel) => {
+        const channelId = channel.id;
+        const membership = memberships[channelId];
+        if (!isChannelMuted(membership)) {
+            if (membership.mention_count > 0) {
+                return {
+                    ...acc,
+                    1: [...acc[1], channelId],
+                };
+            } else if (membership.notify_props && membership.notify_props.mark_unread !== 'mention' && channel.total_msg_count - membership.msg_count) {
+                return {
+                    ...acc,
+                    2: [...acc[2], channelId],
+                };
+            }
+        }
+        return acc;
+    }, {
+        1: [], // 1 being high priority requests
```

Comment:
Maybe this should be a priorityEnum type.

Reply:
Would it work? This is not an auto-incremented field. I am ignorant about this and I couldn't find any example using either.

Reply:
Enums values are by default numeric, I guess the benefit I had in mind is that you can express this in the value and not with comments. I am 1/5 on this.

Reply:
Yeah, you could define a TypeScript enum for this like
```typescript
enum Priority {
    High: 1,
    Medium: 2,
    Low: 3,
}
```

and then refer to them by name. It also helps since you can use `Priority` as a type elsewhere instead of our JS-based enums where you have to define a bunch of constants for them.